quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCodeActions.java

```diff
@@ -111,6 +118,50 @@ private void doCodeActionsForUnknown(Diagnostic diagnostic, PropertiesModel docu
 		}
 	}
 
+	/**
+	 * Create code action fo 
+	 * @param diagnostic
+	 * @param document
+	 * @param projectInfo
+	 * @param valuesRulesManager
+	 * @param codeActions
+	 */
+	private void doCodeActionsForUnknownEnumValue(Diagnostic diagnostic, PropertiesModel document,
+			QuarkusProjectInfo projectInfo, ValuesRulesManager valuesRulesManager, List<CodeAction> codeActions) {
+		try {
+			PropertyValue propertyValue = (PropertyValue) document.findNodeAt(diagnostic.getRange().getStart());
+			PropertyKey propertyKey = ((Property) propertyValue.getParent()).getKey();
+			String value = propertyValue.getValue();
+			String propertyName = propertyKey.getPropertyName();
+
+			ExtendedConfigDescriptionBuildItem metaProperty = projectInfo.getPropertyFromName(propertyName);
+			if (metaProperty == null) {
+				return;
+			}
+
+			List<EnumItem> enums = metaProperty.getEnums();

```

Comment:
In QuarkusCompletions, you have a private static method

```java
 Collection<EnumItem> getEnums(ExtendedConfigDescriptionBuildItem property, PropertiesModel model,
			ValuesRulesManager valuesRulesManager) 
```
which does the same thing (nut manages boolean type too).

I suggest you to move this private static method from QuarkusCompletion in the QuarkusPropertiesUtils and you consume it here. 
Replace this enums collection by creating a new method utility `Collection<EnumItem> getEnums(ExtendedConfigDescriptionBuildItem property, PropertiesModel model,	ValuesRulesManager valuesRulesManager)` You can find this method in